File: intellex-backend/backend/app/api/schemas/ai.py
from typing import Literal

from pydantic import BaseModel


class AIStatusResponse(BaseModel):
    configured: bool
    model: str


class AIChatTurn(BaseModel):
    role: Literal["user", "assistant"]
    content: str


class AIChatRequest(BaseModel):
    question: str
    history: list[AIChatTurn] = []


class AISourceResponse(BaseModel):
    id: str
    title: str
    url: str
    source: str


class AIChatResponse(BaseModel):
    answer: str
    sources: list[AISourceResponse]
    model: str

File: intellex-frontend/frontend/src/lib/api/ai.ts
import { apiClient } from "@/lib/api/client";
import type { AISource, AIStatus } from "@/lib/types";

export async function getAIStatus(): Promise<AIStatus> {
  const { data } = await apiClient.get<{ configured: boolean; model: string }>(
    "/ai/status"
  );

  return data;
}

export interface AskAIParams {
  question: string;
  history?: Array<{ role: "user" | "assistant"; content: string }>;
}

export interface AskAIResult {
  answer: string;
  sources: AISource[];
  model: string;
}

export async function askAI(params: AskAIParams): Promise<AskAIResult> {
  const { data } = await apiClient.post<AskAIResult>(
    "/ai/chat",
    {
      question: params.question,
      history: params.history ?? [],
    },
    { timeout: 45_000 }
  );

  return data;
}
